### PR TITLE
s3select: multi-rows tests

### PR DESCRIPTION
**Pull request by albin-antony**
Migrate queries from s3-tests

Signed-off-by: Albin Antony <aantony@redhat.com>

### CODE REVIEW

**Comment by galsalomon66** [comment]
need to add assert for failure, for one of the two queries.
in case comparing the results of 2 queries and both failed, there is no indication for that. (it will pass with success)

as discussed, to add the rest of the tests.

File: test/s3select_test.cpp [hunk 1]
```diff
@@ -2880,3 +2897,274 @@ TEST(TestS3selectFunctions, null_column )
...
+
+  const std::string input_query_6 = "select count(0) from  stdin where date_diff(hour,utcnow(),date_add(day,1,utcnow())) == 24;";
+
+  std::string s3select_result_6 = run_s3select(input_query_6,input);
+
+  ASSERT_EQ(s3select_result_6, s3select_result_4);
+}
+
+TEST(TestS3selectFunctions, test_like_expressions)
+{
```
Comment (galsalomon66) [inline test/s3select_test.cpp:3126]: in s3test there are more tests for the /like/ operator.

**Comment by galsalomon66** [comment]
there is no multi-row tests for trim operator

File: test/s3select_test.cpp [hunk 2]
```diff
@@ -2880,3 +2915,447 @@ TEST(TestS3selectFunctions, null_column )
...
+
+  ASSERT_NE(s3select_result_1,failure_sign);
+
+  ASSERT_EQ(s3select_result_1, "ab12cd$$,\n");
+
+  const std::string input_query_2 = "select upper(\"ab12CD$$\") from s3object;";
+
+  std::string s3select_result_2 = run_s3select(input_query_2,input);
+
+  ASSERT_NE(s3select_result_2,failure_sign);
```
Comment (galsalomon66) [inline test/s3select_test.cpp:2992]: one assert is enough

File: test/s3select_test.cpp [hunk 3]
```diff
@@ -2880,3 +2915,447 @@ TEST(TestS3selectFunctions, null_column )
...
+
+  std::string s3select_result_3 = run_s3select(input_query_3,input);
+
+  ASSERT_NE(s3select_result_3,failure_sign);
+
+  const std::string input_query_4 = "select count(*) from s3object;";
+
+  std::string s3select_result_4 = run_s3select(input_query_4,input);
+
+  ASSERT_NE(s3select_result_4,failure_sign);
```
Comment (galsalomon66) [inline test/s3select_test.cpp:3164]: no need assert for /s3select_result_4/

File: test/s3select_test.cpp [hunk 4]
```diff
@@ -2880,3 +2915,447 @@ TEST(TestS3selectFunctions, null_column )
...
+
+  ASSERT_EQ(s3select_result_3, s3select_result_4);
+
+  const std::string input_query_5 = "select count(0) from  stdin where date_diff(year,to_timestamp(_1),date_add(day, 366 ,to_timestamp(_1))) == 1;";
+
+  std::string s3select_result_5 = run_s3select(input_query_5,input);
+
+  ASSERT_EQ(s3select_result_5, s3select_result_4);
```
Comment (galsalomon66) [inline test/s3select_test.cpp:3172]: missing /assert failure_sign/ for s3select_result_5 or s3select_result_6

File: test/s3select_test.cpp [hunk 5]
```diff
@@ -2880,3 +2915,447 @@ TEST(TestS3selectFunctions, null_column )
...
+
+  std::string s3select_result_1 = run_s3select(input_query_1,input);
+
+  ASSERT_NE(s3select_result_1,failure_sign);
+
+  const std::string input_query_2 = "select count(*) from stdin where substring(_1 from 6 for 5) == \"aeiou\";";
+
+  std::string s3select_result_2 = run_s3select(input_query_2,input);
+
+  ASSERT_NE(s3select_result_2,failure_sign);
```
Comment (galsalomon66) [inline test/s3select_test.cpp:3344]: can remove this one

File: test/s3select_test.cpp [hunk 6]
```diff
@@ -2880,3 +2915,447 @@ TEST(TestS3selectFunctions, null_column )
...
+
+  std::string s3select_result_3 = run_s3select(input_query_3,input);
+
+  ASSERT_NE(s3select_result_3,failure_sign);
+
+  const std::string input_query_4 = "select count(*) from stdin where substring(_1,6,5) == \"aeiou\";";
+
+  std::string s3select_result_4 = run_s3select(input_query_4,input);
+
+  ASSERT_NE(s3select_result_4,failure_sign);
```
Comment (galsalomon66) [inline test/s3select_test.cpp:3358]: can remove this one

**Comment by galsalomon66** [comment]
redundant assert, please remove these

File: test/s3select_test.cpp (hunk 6 above)
Comment (albin-antony) [reply test/s3select_test.cpp]: Removed

**Pull request merged**